core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1987,24 +1985,33 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
 
-            DBIO.from(samDAO.createResourceFull(SamResourceTypeNames.workspace, workspaceId, defaultPolicies, workspaceRequest.authorizationDomain.getOrElse(Set.empty).map(_.membersGroupName.value), userInfo)).map(_ => defaultPolicies)
+            DBIO.from(samDAO.createResourceFull(SamResourceTypeNames.workspace, workspaceId, defaultPolicies, workspaceRequest.authorizationDomain.getOrElse(Set.empty).map(_.membersGroupName.value), userInfo))
           }
-          _ <- DBIO.from(createWorkflowCollectionForWorkspace(workspaceId))
-          _ <- DBIO.from(Future.traverse(policies.toSeq) { case (policyName, _) =>
-            if (policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
-              // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
-              // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
-              // the limit of 2000
-              Future.successful(())
-            } else if (WorkspaceAccessLevels.withPolicyName(policyName.value).isDefined) {
-              // only sync policies that have corresponding WorkspaceAccessLevels to google because only those are
-              // granted bucket access (and thus need a google group)
-              samDAO.syncPolicyToGoogle(SamResourceTypeNames.workspace, workspaceId, policyName)
-            } else {
-              Future.successful(())
+
+          // policyMap has policyName -> policyEmail
+          policyMap: Map[SamResourcePolicyName, WorkbenchEmail] = resource.accessPolicies.map( x => SamResourcePolicyName(x.id.accessPolicyName) -> WorkbenchEmail(x.email)).toMap
+
+          _ <- DBIO.from(createWorkflowCollectionForWorkspace(workspaceId, policyMap))
+
+          _ <- DBIO.from(
+            Future.traverse(policyMap) { x =>
+              val policyName = x._1
+              if (policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
+                // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
+                // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
+                // the limit of 2000
+                Future.successful(())
+              } else if (WorkspaceAccessLevels.withPolicyName(policyName.value).isDefined) {
+                // only sync policies that have corresponding WorkspaceAccessLevels to google because only those are
+                // granted bucket access (and thus need a google group)
+                samDAO.syncPolicyToGoogle(SamResourceTypeNames.workspace, workspaceId, policyName)
+                Future.successful(())
```

Comment:
this looks like a functional change. Previously this code path returned the value of `syncPolicyToGoogle` ... now it is hardcoded to be successful. Did you intend this?

Reply:
no -- excellent catch!!